tests/test_filer.py

```diff
@@ -199,11 +205,172 @@ class FilerTest_no_env(unittest.TestCase, AssertThrowsMixin):
 
     def test_newTransput_file_disabled(self):
 
-        self.assertThrows( lambda: newTransput('file')
+        self.assertThrows( lambda: newTransput('file','/home/user/test')
                          , FileProtocolDisabled
                          , "'file:' protocol disabled\n"
                            "To enable it, both 'HOST_BASE_PATH' and 'CONTAINER_BASE_PATH' environment variables must be defined."
                          )
+class S3_FilerTest(unittest.TestCase,AssertThrowsMixin):
+
+    def setUp(self):
+        self.mock = mock_s3()
+        self.mock.start()
+        # moto package fails if localhost is given as endpoint_url
+        self.conn = boto3.resource('s3', endpoint_url=None)
+        self.conn.create_bucket(Bucket='tesk')
+        path = "/home/user/filer_test/"
+        url = "http://s3.localhost:9000/tesk/test/bucket_file1.txt"
+        ftype = "FILE"
+        self.trans = S3Transput(path, url, ftype)
+
+    def tearDown(self):
+        self.mock.stop()
+
+    def test_get_names(self):
+        """
+        Check if the bucket name and path is extracted
+        """
+        path = "/home/user/filer_test/"
+        url = "http://mybucket.s3-aws-region.amazonaws.com/folder/file.txt"
+        ftype = "FILE"
+        self.trans = S3Transput(path, url, ftype)
+        self.assertEqual(self.trans.get_names(), ('mybucket', '/folder/file.txt'))
```

Comment:
This is just one test of one URL style. Can you please provide full coverage for all cases that you support?
What does `get_names` return for a directory?

Reply:
Updated with more input urls.